synapse/util/async_helpers.py

```diff
@@ -496,8 +496,8 @@ def time_it_out():
 
         try:
             deferred.cancel()
-        except:  # noqa: E722, if we throw any exception it'll break time outs
-            logger.exception("Canceller failed during timeout")
+        except Exception as e:  # if we throw any exception it'll break time outs
+            logger.exception("Canceller failed during timeout", exc_info=e)
```

Comment:
Doesn't `logger.exception` pull this automatically? It seems better to not declare it if we don't have to.

Reply:
I like explicitness over implicitness, i think this is easier to read.

Same to the other comment.

Though if this is really not desired, I can change this back.

Reply:
Please change it back.

I get the explicit vs. implicit argument, but it also adds fragility in case APIs ever change. (No, I don't really think this API will change, but philosophically if we can avoid passing things in it avoids brittleness and is easier to read.)

Reply:
I'll change it back. 👍

I don't think python really wants to emphasize itself around implicitness, but I do have to agree that with the presence of twisted in the codebase (which does a lot of implicit magic), things are a lot more fragile.

In general, i think python is mainly built around being explicit first and foremost, so that's where my intention and argument comes from.